amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/TestConsumer.java

```diff
@@ -188,23 +188,42 @@ private ByteBuffer wrapWithCounter(int payloadSize, BigInteger payloadCounter) t
         return ByteBuffer.wrap(returnData);
     }
 
-    private void deleteResources(StreamExistenceManager streamExistenceManager, DynamoDbAsyncClient dynamoDBClient) throws Exception {
-        log.info("-------------Start deleting stream.----------------");
-        streamExistenceManager.deleteStream(this.streamName);
-        log.info("-------------Start deleting lease table.----------------");
-        deleteLeaseTable(dynamoDBClient, consumerConfig.getStreamName());
-        log.info("-------------Finished deleting resources.----------------");
+    private void stopProducer() {
+        log.info("Cancelling producer and shutting down executor.");
+        producerFuture.cancel(false);
+        producerExecutor.shutdown();
     }
 
-    private void deleteLeaseTable(DynamoDbAsyncClient dynamoClient, String tableName) throws Exception {
-        DeleteTableRequest request = DeleteTableRequest.builder().tableName(tableName).build();
+    private void awaitConsumerFinish() throws Exception {
+        Future<Boolean> gracefulShutdownFuture = scheduler.startGracefulShutdown();
+        log.info("Waiting up to 20 seconds for shutdown to complete.");
         try {
-            FutureUtils.resolveOrCancelFuture(dynamoClient.deleteTable(request), Duration.ofSeconds(60));
-        } catch (ExecutionException e) {
-            throw new Exception("Could not delete lease table: {}", e);
+            gracefulShutdownFuture.get(20, TimeUnit.SECONDS);
         } catch (InterruptedException e) {
-            throw new Exception("Deleting lease table interrupted: {}", e);
+            log.info("Interrupted while waiting for graceful shutdown. Continuing.");
+        } catch (ExecutionException | TimeoutException e) {
+            throw new Exception("Exception while executing graceful shutdown. {}", e);
         }
+        log.info("Completed, shutting down now.");
+    }
 
+    private void validateRecordProcessor() throws Exception {
+        log.info("The number of expected records is: {}", successfulPutRecords);
+        final RecordValidationStatus errorVal = consumerConfig.getRecordValidator().validateRecords(successfulPutRecords);
+        if (errorVal == RecordValidationStatus.OUT_OF_ORDER) {
+            throw new RuntimeException("There was an error validating the records that were processed. The records were out of order");
+        } else if (errorVal == RecordValidationStatus.MISSING_RECORD) {
+            throw new RuntimeException("There was an error validating the records that were processed. Some records were missing.");
+        }
```

Comment:
combine into one ?

Reply:
I think it is helpful to have a message telling which one of these two errors occurred because both are checked in the validation function. If you do not think so, I can change that function to just return true (success) or false (fail).